### PR TITLE
Add remaining onboarding documentation

### LINKED ISSUE #3198: Onboarding: Documentation and user flow
As noted by @timmyc, we should document how the new onboarding experience works and how the various pieces of onboarding connect between the store and other sites.

* New REST API endpoints.
* Onboarding filters and hooks.
* WCCOM connection and redirection
* Calypso and Jetpack Connections
* Building the onboarding feature plugin.

### PR DESCRIPTION
**Pull request by justinshreve**
Closes #3198. This PR adds the rest if the onboarding dev documentation outlined in that issue. 

### Detailed test instructions:

* Review the onboard.md changes. [GitHub View](https://github.com/woocommerce/woocommerce-admin/blob/3d3c7d23a4ab0801cde3db806fc8e42caf193181/docs/features/onboarding.md).

### CODE REVIEW

File: docs/features/onboarding.md [hunk 1]
```diff
@@ -19,35 +19,107 @@ You can also set the following configuration flag in your `wp-config.php`:
...
+* `rest_onboarding_profile_collection_params` filters the collection parameters for requests to  `/wc-admin/onboarding/profile`.
+* `woocommerce_admin_onboarding_industries` filters the list of allowed industries displayed in the profile wizard.
+* `woocommerce_admin_onboarding_industry_image` filters the images used for homepage templates in the appearance task. When creating a homepage, example images are used based on industry. These images are stored in `images/onboarding`.
+* `woocommerce_admin_onboarding_product_types` filters the product types displayed in the profile wizard.
+* `woocommerce_onboarding_plugins_whitelist` filters the list of plugins that can installed & activated via onboarding. This acts as a whitelist so only certain plugins can be used via the `/wc-admin/onboarding/profile/install` and `/wc-admin/onboarding/profile/activate` endpoints.
+* `woocommerce_admin_onboarding_themes` filters the themes displayed in the profile wizard.
+* `woocommerce_onboarding_jetpack_connect_redirect_url` filters the Jetpack connection URL outlined in the Jetpack connection section below.
```
Comment (joshuatf) [inline docs/features/onboarding.md]: Small change:

```suggestion
* `woocommerce_onboarding_jetpack_connect_redirect_url` filters the Jetpack connection redirect URL outlined in the Jetpack connection section below.
```

File: docs/features/onboarding.md [hunk 2]
```diff
@@ -19,35 +19,107 @@ You can also set the following configuration flag in your `wp-config.php`:
...
+* POST `/wc-admin/onboarding/plugins/connect-square` - Generates a URL for connecting to Square during the payments task.
+* GET `/wc-admin/onboarding/profile` - Returns the information gathered during the profile wizard. See the `woocommerce_onboarding_profile_properties` array for a list of fields.
+* POST `/wc-admin/onboarding/profile` - Sets data for the profile wizard. See the `woocommerce_onboarding_profile_properties` array for a list of fields.
+* POST `/wc-admin/onboarding/tasks/import_sample_products` - Used for importing sample products during the appearance task.
+* POST `/wc-admin/onboarding/tasks/create_store_pages` - Used for creating default store pages (like my account and checkout) during the profile wizard.
+* POST `/wc-admin/onboarding/tasks/create_homepage` - Used for creating a homepage using Gutenberg templates.
+
+## Onboarding filters
+
+* `woocommerce_onboarding_profile_properties` filters the properties we track as part of the profile wizard (such as information from store/business details steps).
```
Comment (joshuatf) [inline docs/features/onboarding.md]: Should we mention somewhere that if the `completed` property is set to `true` this hides the profiler?

File: docs/features/onboarding.md [hunk 3]
```diff
@@ -19,35 +19,107 @@ You can also set the following configuration flag in your `wp-config.php`:
...
-
-@todo
 
 ## Options and settings
 
-@todo
+A few new WordPress options have been introduced to store information and settings during setup. It may be necessary to manual delete these options from your `wp_options` database to test a certain task or feature.
 
-## WooCommerce.com connection and redirection
+* `woocommerce_onboarding_payments`. Since the payments step requires multiple redirects to payment providers to setup accounts, we cache the current progress of the payments step in an option, so that we can quickly drop users back into the correct part of the task.
```
Comment (joshuatf) [inline docs/features/onboarding.md]: We can also add `woocommerce_task_list_hidden` to this list.

Comment (justinshreve) [reply docs/features/onboarding.md]: Thanks for catching that -- definitely one of the more important ones to know about!

**Comment by justinshreve** [comment]
Thanks for the review! Handled your feedback.

> Unrelated, but one thing I feel we need to do after seeing this is come up with consistent naming patterns for filters across wc-admin.

I'll log an issue for handing these on the onboarding side. I feel like we can still do this for fresh onboarding code without causing issues.

**Pull request merged**